### PR TITLE
[#4927]-remove generic resource

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Removal of the GenericResource type should not affect any functionalities of hydroshare.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4928](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4928)

File: hs_composite_resource/serialization.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+from hs_core.serialization import GenericResourceMeta
+
+
+class CompositeResourceMeta(GenericResourceMeta):
```
Comment (devincowan) [inline hs_composite_resource/serialization.py:5]: Is it a bad idea to rename GenericResourceMeta ->  CompositeResourceMeta throughout the codebase instead? More or less confusing?

**Review by devincowan — APPROVED** [review]
sweet!
a few additional places that might or might not need changes:

1. This doc:
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/docs/access-control.md?plain=1#L14
2. Do we want to rename this file and its imports? https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_collection_resource/templates/pages/collectionresource.html#L1
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_composite_resource/templates/pages/compositeresource.html#L3
3. I don't think we need / want to modify, but just mentioning these migrations:
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/migrations/0001_initial.py#L150
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/migrations/0002_genericresource_resource_type.py#L15
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/migrations/0004_auto_20150721_1125.py#L15
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/migrations/0006_auto_20150917_1515.py#L15
4. This template https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/templates/pages/share-resource.html#L12
5. Comments in this file
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/native/test_copy_resource.py#L44
6. Test name
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/rest/test_resource_science_meta.py#L42
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/rest/test_resource_science_meta.py#L143
7. comment
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_tools_resource/utils.py#L14
8. there are sundry tests that still have "Generic ..." used for values. I don't think we really need to change these. A lot of them are resource TITLEs. An example:
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/native/test_core_metadata.py#L59

File: hs_core/models.py [hunk 2]
```diff
@@ -4041,25 +4041,6 @@ def find_non_preferred_folder_paths(dir_path):
         return not_preferred_paths
 
 
-# TODO Deprecated
-class GenericResource(BaseResource):
```
[the remark this answers is not in the document]
Comment (pkdash) [reply hs_core/models.py:4045]: These are in BaseResource and are overridden in all sub classes - so I think no need to change.

File: hs_composite_resource/serialization.py (hunk 1 above)
Comment (devincowan) [reply hs_composite_resource/serialization.py:5]: ah good point, thanks for contemplating. Let's stick with what you have

**Review by devincowan — APPROVED** [review]

**Comment by pkdash** [comment]
@devincowan See if I have addressed your suggestions.

**Review by devincowan — DISMISSED** [review]
I think we are good! There are still some "Generic" references, but nothing that is too confusing I think...

Access control documentation:
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/docs/access-control.md?plain=1#L14

Comments in tests, for example (but not limited to):
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/native/test_copy_resource.py#L62
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/native/test_copy_resource.py#L75

Names in tests, for example (but not limited to):
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/native/test_core_metadata.py#L1380
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/native/test_core_metadata.py#L1380
https://github.com/hydroshare/hydroshare/blob/4927-remove-generic-resource/hs_core/tests/api/native/test_core_metadata.py#L2552

I'm approving without these modifications because I think they are just icing on the cake. If you want, I would be happy to push a commit that bulk changes them.

**Review by devincowan — APPROVED** [review]

**Pull request merged**